### ISSUE TITLE
bones-check assumes files are generated.

### ISSUE THREAD

**Issue by bmorcos** [nengo/nengo-bones#61]
I'm setting up `nengo-bones` for `nengo-pynq`. I am using a manually written `docs/conf.py` and thus omitting `docs_conf_py` from `nengobones.yml`. Despite this, `bones-check` throws an error since it tries to validate the generated file:

```
**************************************************
Checking content of nengo-bones generated files:
root dir: .

.travis.yml:
  Up to date
.codecov.yml:
  Up to date
docs/conf.py:
Traceback (most recent call last):
  File "/home/ben/anaconda3/bin/bones-check", line 11, in <module>
    load_entry_point('nengo-bones', 'console_scripts', 'bones-check')()
  File "/home/ben/anaconda3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/ben/anaconda3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/ben/anaconda3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ben/anaconda3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/ben/git/nengo-bones/nengo_bones/scripts/check_bones.py", line 59, in main
    new_lines = template.render(**template.get_render_data(config)).splitlines(
  File "/home/ben/git/nengo-bones/nengo_bones/templates.py", line 95, in get_render_data
    data.update(config[self.section])
KeyError: 'docs_conf_py'
```

@drasmuss noted " it looks like it assumes that if the file is there, it was generated by nengo-bones (so you can skip having the file, but can't have the file and not have it be templated)"

As a workaround, I've created `.templates/docs/conf.py.template`, but it would be nice to not require this and instead to recognize that this option is not set in `nengobones.yml`.

**Comment by bmorcos** [comment]
This may be specific to `docs/conf.py` actually, I just removed the templated license from `nengo-pynq` and it correctly notes that it was not generated from `nengo-bones`:

```
8:49:56 ben@phoenix ~/Git/nengo-pynq bones*? ∴ bones-check
**************************************************
Checking content of nengo-bones generated files:
root dir: .

.travis.yml:
  Up to date
.codecov.yml:
  Up to date
docs/conf.py:
  Up to date
setup.cfg:
  Up to date
setup.py:
  Up to date
CONTRIBUTING.rst:
  Up to date
CONTRIBUTORS.rst:
  Up to date
LICENSE.rst:
  This file was not generated with nengo-bones
MANIFEST.in:
  File not found
.pre-commit-config.yaml:
  Up to date
pyproject.toml:
  Up to date
**************************************************
```

**Comment by tbekolay** [comment]
So I looked into this issue, and what I've deduced happened is that the `docs/conf.py` file that was raising the error still contained the text

```
Automatically generated by nengo-bones
```

Because of that, we looked in the .nengobones.yml configuration for its information so that we could check if it matched what it should look like, hence the key error. If the file had not contained that text, we would have raised the same error as `LICENSE.rst` in the most recent comment. So, it wasn't really specific to the handling of `docs/conf.py`, but specific to the contents of @bmorcos's `docs/conf.py`.

I made #66 to handle this case more gracefully, by printing a hopefully sufficient error message and failing `bones-check` (as this is definitely not something we want to allow).
